File: src/java/magma/api/Tuple2Impl.java
package magma.api;

public record Tuple2Impl<A, B>(A leftValue, B rightValue) implements Tuple2<A, B> {
    @Override
    public A left() {
        return this.leftValue;
    }

    @Override
    public B right() {
        return this.rightValue;
    }
}

File: src/windows/magma/api/Tuple2Impl.c
#include "./Tuple2Impl.h"
export class Tuple2Impl<A, B> implements Tuple2<A, B> {
	leftValue: A;
	rightValue: B;
	constructor (leftValue: A, rightValue: B) {
		this.leftValue = leftValue;
		this.rightValue = rightValue;
	}
}

left(): A {
	return this.leftValue;
}
right(): B {
	return this.rightValue;
}